### PR TITLE
7709: Added the check for return type of len function.

### PR DESCRIPTION
**Pull request by chetankhilosiya**

### CODE REVIEW

File: crates/hir/src/lib.rs [hunk 1]
```diff
@@ -1631,6 +1631,13 @@ impl Type {
         matches!(self.ty.value.interned(&Interner), TyKind::Ref(hir_ty::Mutability::Mut, ..))
     }
 
+    pub fn is_usize(&self) -> bool {
+        matches!(
+            self.ty.value.interned(&Interner),
+            TyKind::Scalar(Scalar::Uint(hir_ty::primitive::UintTy::Usize))
```
Comment (Veykril) [inline crates/hir/src/lib.rs:1637]: Let's import `hir_ty::primitive::UintTy` here so we can use `UintTy::Usize`

**Comment by Veykril** [comment]
bors d+

**Comment by lnicola** [comment]
changelog fixup #8037

**Pull request merged**